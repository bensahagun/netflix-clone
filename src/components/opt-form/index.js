import React from 'react';
import { Container, Input, Button, Break, Text } from './styles/opt-form';

export default function OptForm({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

OptForm.Input = function ({ ...restProps }) {
  return <Input {...restProps}></Input>;
};

OptForm.Button = function ({ children, ...restProps }) {
  return (
    <Button {...restProps}>
      {children} <img src='/images/icons/chevron-right.png' alt='Try now' />
    </Button>
  );
};

OptForm.Text = function ({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

OptForm.Break = function ({ ...restProps }) {
  return <Break {...restProps} />;
};
