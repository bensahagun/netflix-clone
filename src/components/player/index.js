import React, { useState, useContext, createContext } from 'react';
import ReactDOM from 'react-dom';
import { Container, Button, FeatureButton, Overlay, Inner, Close } from './styles/player';

export const PlayerContext = createContext();

export default function Player({ children, ...restProps }) {
  const [showPlayer, setShowPlayer] = useState(false);

  return (
    <PlayerContext.Provider value={{ showPlayer, setShowPlayer }}>
      <Container {...restProps}>{children}</Container>
    </PlayerContext.Provider>
  );
}

Player.Video = function PlayerVideo({ src, ...restProps }) {
  const { showPlayer, setShowPlayer } = useContext(PlayerContext);

  return showPlayer
    ? ReactDOM.createPortal(
        <Overlay onClick={() => setShowPlayer(false)} {...restProps}>
          <Inner>
            <video id='netflix-player' controls autoPlay='autoplay'>
              <source src={src} type='video/mp4' />
            </video>
          </Inner>
          <Close />
        </Overlay>,
        document.body
      )
    : null;
};

Player.Button = function PlayerButton({ ...restProps }) {
  // eslint-disable-next-line no-unused-vars
  const { showPlayer, setShowPlayer } = useContext(PlayerContext);

  return (
    <Button {...restProps} onClick={() => setShowPlayer((showPlayer) => !showPlayer)}>
      Play
    </Button>
  );
};

Player.FeatureButton = function PlayerFeatureButton({ ...restProps }) {
  // eslint-disable-next-line no-unused-vars
  const { showPlayer, setShowPlayer } = useContext(PlayerContext);

  return (
    <FeatureButton {...restProps} onClick={() => setShowPlayer((showPlayer) => !showPlayer)}>
      Play Now
    </FeatureButton>
  );
};
