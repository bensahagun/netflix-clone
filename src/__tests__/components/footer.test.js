import React from 'react';
import { render } from '@testing-library/react';
import { Footer } from '../../components';

describe('<Footer/>', () => {
  it('renders <Footer/> with data', () => {
    const { container, getByText } = render(
      <Footer>
        <Footer.Title>Questions? Contact us.</Footer.Title>
        <Footer.Break />
        <Footer.Row>
          <Footer.Column>
            <Footer.Link href='#'>FAQ</Footer.Link>
            <Footer.Link href='#'>Investor Relations</Footer.Link>
            <Footer.Link href='#'>Ways to Watch</Footer.Link>
            <Footer.Link href='#'>Corporate Information</Footer.Link>
            <Footer.Link href='#'>Netflix Originals</Footer.Link>
          </Footer.Column>
          <Footer.Column>
            <Footer.Link href='#'>Help Centre</Footer.Link>
            <Footer.Link href='#'>Jobs</Footer.Link>
            <Footer.Link href='#'>Terms of Use</Footer.Link>
            <Footer.Link href='#'>Contact Us</Footer.Link>
          </Footer.Column>
          <Footer.Column>
            <Footer.Link href='#'>Account</Footer.Link>
            <Footer.Link href='#'>Redeem Gift Cards</Footer.Link>
            <Footer.Link href='#'>Privacy</Footer.Link>
            <Footer.Link href='#'>Speed Test</Footer.Link>
          </Footer.Column>
          <Footer.Column>
            <Footer.Link href='#'>Media Centre</Footer.Link>
            <Footer.Link href='#'>Buy Gift Cards</Footer.Link>
            <Footer.Link href='#'>Cookie Preferences</Footer.Link>
            <Footer.Link href='#'>Legal Notices</Footer.Link>
          </Footer.Column>
        </Footer.Row>
        <Footer.Break />
        <Footer.Text>Netflix United States</Footer.Text>
      </Footer>
    );

    expect(getByText('Questions? Contact us.')).toBeTruthy();
    expect(getByText('FAQ')).toBeTruthy();
    expect(getByText('Investor Relations')).toBeTruthy();
    expect(getByText('Ways to Watch')).toBeTruthy();
    expect(getByText('Corporate Information')).toBeTruthy();
    expect(getByText('Help Centre')).toBeTruthy();
    expect(getByText('Jobs')).toBeTruthy();
    expect(getByText('Terms of Use')).toBeTruthy();
    expect(getByText('Contact Us')).toBeTruthy();
    expect(getByText('Account')).toBeTruthy();
    expect(getByText('Redeem Gift Cards')).toBeTruthy();
    expect(getByText('Privacy')).toBeTruthy();
    expect(getByText('Speed Test')).toBeTruthy();
    expect(getByText('Media Centre')).toBeTruthy();
    expect(getByText('Buy Gift Cards')).toBeTruthy();
    expect(getByText('Cookie Preferences')).toBeTruthy();
    expect(getByText('Legal Notices')).toBeTruthy();
    expect(getByText('Netflix United States')).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
