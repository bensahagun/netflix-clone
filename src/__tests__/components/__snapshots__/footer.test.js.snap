// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> renders <Footer/> with data 1`] = `
<div
  class="footer__Container-sc-qu6i6-0 eGlRgl"
>
  <p
    class="footer__Title-sc-qu6i6-4 jjOPAg"
  >
    Questions? Contact us.
  </p>
  <div
    class="footer__Break-sc-qu6i6-6 iyNlEP"
  />
  <div
    class="footer__Row-sc-qu6i6-2 ctgCBW"
  >
    <div
      class="footer__Column-sc-qu6i6-1 jwsorJ"
    >
      <a
        class="footer__Link-sc-qu6i6-3 ezAywl"
        href="#"
      >
        FAQ
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 ezAywl"
        href="#"
      >
        Investor Relations
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 ezAywl"
        href="#"
      >
        Ways to Watch
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 ezAywl"
        href="#"
      >
        Corporate Information
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 ezAywl"
        href="#"
      >
        Netflix Originals
      </a>
    </div>
    <div
      class="footer__Column-sc-qu6i6-1 jwsorJ"
    >
      <a
        class="footer__Link-sc-qu6i6-3 ezAywl"
        href="#"
      >
        Help Centre
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 ezAywl"
        href="#"
      >
        Jobs
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 ezAywl"
        href="#"
      >
        Terms of Use
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 ezAywl"
        href="#"
      >
        Contact Us
      </a>
    </div>
    <div
      class="footer__Column-sc-qu6i6-1 jwsorJ"
    >
      <a
        class="footer__Link-sc-qu6i6-3 ezAywl"
        href="#"
      >
        Account
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 ezAywl"
        href="#"
      >
        Redeem Gift Cards
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 ezAywl"
        href="#"
      >
        Privacy
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 ezAywl"
        href="#"
      >
        Speed Test
      </a>
    </div>
    <div
      class="footer__Column-sc-qu6i6-1 jwsorJ"
    >
      <a
        class="footer__Link-sc-qu6i6-3 ezAywl"
        href="#"
      >
        Media Centre
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 ezAywl"
        href="#"
      >
        Buy Gift Cards
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 ezAywl"
        href="#"
      >
        Cookie Preferences
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 ezAywl"
        href="#"
      >
        Legal Notices
      </a>
    </div>
  </div>
  <div
    class="footer__Break-sc-qu6i6-6 iyNlEP"
  />
  <p
    class="footer__Text-sc-qu6i6-5 eipPUF"
  >
    Netflix United States
  </p>
</div>
`;
